android_jni/avifandroidjni/src/main/java/com/avif/android/AvifDecoder.java

```diff
@@ -0,0 +1,58 @@
+package com.avif.android;
+
+import android.graphics.Bitmap;
+import java.nio.ByteBuffer;
+
+/** An AVIF Decoder. AVIF Specification: https://aomediacodec.github.io/av1-avif/. */
+@SuppressWarnings("CatchAndPrintStackTrace")
+public class AvifDecoder {
+  static {
+    try {
+      System.loadLibrary("avif_android");
+    } catch (UnsatisfiedLinkError exception) {
+      exception.printStackTrace();
+    }
+  }
+
+  // This is a utility class and cannot be instantiated.
+  private AvifDecoder() {}
+
+  /** Contains information about the AVIF Image dimensions. */
+  public static class Info {
+    public int width;
+    public int height;
+  }
+
+  /**
+   * Returns true if the bytes in the buffer seem like an AVIF image.
+   *
+   * @param buffer The encoded image.
+   * @return true if the bytes seem like an AVIF image, false otherwise.
+   */
+  public static boolean isAvifStream(ByteBuffer buffer) {
+    return AvifDecoder.isAvifStream(buffer, buffer.remaining());
+  }
+
+  private static native boolean isAvifStream(ByteBuffer encoded, int length);
+
+  /**
+   * Parses the AVIF header and populates the Info.
+   *
+   * @param encoded The encoded AVIF image.
+   * @param length Length of the encoded buffer.
```

Comment:
Why do we use a separate `length` parameter? The ByteBuffer object can tell us how many bytes of data it contains, right?